ui/src/main/webapp/src/app/components/reports/application-index/application-index.component.ts

```diff
@@ -0,0 +1,223 @@
+import { Component, OnInit, ChangeDetectorRef, ElementRef } from "@angular/core";
+import { Http } from "@angular/http";
+import { Params, ActivatedRoute, Router } from "@angular/router";
+import { utils } from "../../../utils";
+import { NotificationService } from "../../../services/notification.service";
+import { ApplicationGroup } from "windup-services";
+import { AggregatedStatisticsService } from "./aggregated-statistics.service";
+
+@Component({
+    templateUrl: '/application-index.component.html',
+    styleUrls: ['/application-index.component.css']
+})
+export class ApplicationIndexComponent implements OnInit {
+
+    // color scheme for graphs on template
+    colorScheme = {
+        domain: ['#2E4272', '#553900', '#4F628E', '#162955', '#061539', '#AA8439', '#FFE3AA']
+    };
+
+    // both are used for getting graph and for report filter DTO
+    private execID: number;
+    private group: ApplicationGroup;
+
+    // aggregated statistics variables
+    globalPackageUseData: ChartStatistic[] = [];
+    categoriesMultiStats: any[] = [];
+    mandatoryMultiStats: any[] = [];
+
+    constructor(
+        private _router: Router,
+        private _activatedRoute: ActivatedRoute,
+        private _notificationService: NotificationService,
+        private _aggregatedStatsService: AggregatedStatisticsService,
+        private _http: Http
+    ) { }
+
+
+    ngOnInit(): void {
+        this._activatedRoute.params.subscribe(params => {
+            let executionId = parseInt(params['executionId']);
+            this.execID = executionId;
+
+
+            this._aggregatedStatsService.getAggregatedCategories(executionId).subscribe(
+                result => this.categoriesMultiStats = this.getCategoriesStats(this.calculateCategoryIncidents(result), this.calculateStoryPointsInCategories(result)),
+                error => {
+                    this._notificationService.error(utils.getErrorMessage(error))
```

Comment:
I noticed that a few lines like this miss semicolons at the end. That's obviously not a big deal, but it does generate warnings in the IDE.